### PR TITLE
Update acceleration.c

### LINKED ISSUE #2542: 'accSamplingInterval'  is used uninitialized, cause 'accSmooth[]' always zero!
accSamplingInterval <- biquadFilterInitLPF() <- accelerationFilterInit() <- activateConfig <- readEEPROM() <- init() .

### PR DESCRIPTION
**Pull request by andre92**
this should fix #2542

### CODE REVIEW

**Comment by de-o** [comment]
I tested the most recent branch (bf68d28) in @andre92 's fork and it fixes issue #2532.

File: src/main/sensors/acceleration.c [hunk 1]
```diff
@@ -92,6 +92,7 @@ extern bool AccInflightCalibrationActive;
 static flightDynamicsTrims_t *accelerationTrims;
 
 static biquadFilter_t accFilter[XYZ_AXIS_COUNT];
+static bool accFilterInitialised = false;
```
Comment (martinbudden) [inline src/main/sensors/acceleration.c]: `accFilterInitialised` is no longer required, now that `accSamplingInterval` is set.

File: src/main/sensors/acceleration.c [hunk 2]
```diff
@@ -215,7 +216,22 @@ void updateAccelerationReadings(rollAndPitchTrims_t *rollAndPitchTrims)
...
+                accelerationFilterInit(accelerometerConfig()->acc_cut_hz);
+                accFilterInitialised = true;
+            }
+        }
+
+        if (accFilterInitialised) {
+            for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
+                accSmooth[axis] = lrintf(biquadFilterApply(&accFilter[axis], (float)accSmooth[axis]));
+            }
+        }
```
Comment (martinbudden) [inline src/main/sensors/acceleration.c]: These changes are no longer required now that `accSamplingInterval` is set properly.

File: src/main/sensors/acceleration.c [hunk 3]
```diff
@@ -215,7 +215,13 @@ void updateAccelerationReadings(rollAndPitchTrims_t *rollAndPitchTrims)
...
 
     for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
-        accSmooth[axis] = lrintf(accFilterApplyFn(&accFilter[axis], (float)accADCRaw[axis]));
+      accSmooth[axis] = accADCRaw[axis];
+    }
+    
+    if (accelerometerConfig()->acc_cut_hz) {
+        for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
+            accSmooth[axis] = lrintf(biquadFilterApply(&accFilter[axis], (float)accSmooth[axis]));
+        }
```
Comment (hydra) [inline src/main/sensors/acceleration.c]: this shouldn't be needed since the `accFilterApplyFn` will be set appropriately.

**Comment by hydra** [comment]
I made a commit to the master and v1.14.2 branches, please test and report.

**Comment by andre92** [comment]
I just tested angle, horizon and althold in my room on latest commit and both are working (Martian2, SPR F3 EVO, 4S)
may I suggest an emergency bugfix releease?

**Comment by martinbudden** [comment]
@hydra , I agree. This warrants an emergency bugfix release.

**Comment by de-o** [comment]
I built master with this fix merged into it.  On my bench, the acc looks to be fixed.  I'll hopefully get some flight time with it this weekend (fingers crossed for the weather forecast).  Master also contained a merge for preventing a serial overflow while using bl_heli passthrough.  I'll do a quick regression test for it, but not sure how to hit the exact scenario the merge addresses.

**Comment by hydra** [comment]
@andre92 thanks for testing and feedback.

@martinbudden and @andre92 yes I was waiting for feedback before making a new release.

**Comment by hydra** [comment]
I will close this PR as there is nothing left to merge from it.

**Pull request closed without merging**